python/eos/analysis.py

```diff
@@ -155,6 +155,18 @@ def __init__(self, priors, likelihood, global_options={}, manual_constraints={},
         for n in varied_parameter_names - used_parameter_names:
             eos.warn('likelihood does not depend on parameter \'{}\'; remove from prior or check options!'.format(n))
 
+    def __sanitize_manual_input(self, data):
+        """Helper function that cast to float all entries of a manual_constraint"""
+        if type(data) is dict:
+            return { k: self.__sanitize_manual_input(v) for k, v in data.items() }
+        if not np.isscalar(data):
+            return list(map(self.__sanitize_manual_input, data))
+        if np.issubdtype(type(data), int):
+            return int(data)
+        if np.isreal(data):
+            return float(data)
+        return data
```

Comment:
Might be safer to throw an exception here and not return per default at all? We expect the input numbers to be either int or real, right?

Reply:
No, they can be strings (e.g. observable names)

Reply:
In that case, I'd either go with a comment, or better
```
if isinstance(dat, str):
    return data

# all legal cases are covered above
raise ValueError("...")
```

Reply:
I completely agree.